WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift

```diff
@@ -375,6 +383,47 @@ private extension StoreCreationCoordinator {
         analytics.track(event: .StoreCreation.siteCreationStep(step: .profilerCountryQuestion))
     }
 
+    /// This method shows a progress view and proceeds to:
+    /// - Create a simple site
+    /// - Enable Free Trial on the site
+    /// - Wait for JetPack to be installed on the site.
+    ///
+    @MainActor
+    func createFreeTrialStore(from navigationController: UINavigationController, storeName: String) async {
+
+        // Show a progress view while the free trial store is created.
+        showInProgressView(from: navigationController, viewProperties: .init(title: Localization.WaitingForJetpackSite.title, message: ""))
+
+        // Create store site
+        let createStoreResult = await createStore(name: storeName, flow: .wooexpress)
+
+        switch createStoreResult {
+        case .success(let siteResult):
+
+            // Enable Free trial on site
+            let freeTrialResult = await enableFreeTrial(siteID: siteResult.siteID)
+            switch freeTrialResult {
+            case .success:
+                // Wait for jetpack to be installed
+                DDLogInfo("🟢 Free trial enabled on site. Waiting for jetpack to be installed...")
+                waitForSiteToBecomeJetpackSite(from: navigationController, siteID: siteResult.siteID)
+                analytics.track(event: .StoreCreation.siteCreationStep(step: .storeInstallation))
+
+            case .failure(let error):
+                navigationController.popViewController(animated: true)
+                showStoreCreationErrorAlert(from: navigationController, error: SiteCreationError(remoteError: error))
+
+                // TODO: Confirm if we should track a different error here.
+                analytics.track(event: .StoreCreation.siteCreationFailed(source: source.analyticsValue, error: error, flow: .native))
```

Comment:
@wzieba I see this event is fired when a store creation fails, it made sense to me to also fire if the "enable free trial on store" request fails.

Is this ok?